### PR TITLE
Update UG and fix grammar mistake in command result for tag

### PR DESCRIPTION
**Pull request by bchenghi**
- Add glossary
- Add H3 headers to command subsections
- Update command benefits for `sortdesc` and `exit`
- For command result, tag is now tag(s)

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -443,7 +443,21 @@ Action | Format, Examples
 
 <div style="text-align: right"><a href="https://ay2021s1-cs2103t-f13-4.github.io/tp/UserGuide.html">^ Back to top</a></div>
 
-# 7. Glossary
+## 7. Glossary
```
Comment (songyiang) [inline docs/UserGuide.md:446]: Will it be better for the terms in the table to be sorted by alphabetical order?

**Review by songyiang — COMMENTED** [review]
Just a minor suggestion.

File: docs/UserGuide.md (hunk 1 above)
Comment (bchenghi) [reply docs/UserGuide.md:446]: Okay changed, thanks.

**Review by songyiang — APPROVED** [review]
LGTM

**Pull request merged**